job_controller/job_controller.py

```diff
@@ -0,0 +1,80 @@
+"""
+Main class, creates jobs by calling to the jobcreator, creates the jobwatcher for each created job, and recieves
+requests from the topicconsumer.
+"""
+import os
+import threading
+import uuid
+from pathlib import Path
+from typing import Dict
+
+from job_controller.job_watcher import JobWatcher
+from job_controller.job_creator import JobCreator
+from job_controller.script_aquisition import acquire_script
+from job_controller.topic_consumer import TopicConsumer
+from job_controller.utils import create_ceph_path, logger
+
+
+class JobController:
+    """
+    This is the JobController class that will communicate between the consumer and the kubernetes API, it effectively
+    functions as a main class.
+    """
+
+    def __init__(self) -> None:
+        self.ir_api_ip = "irapi.ir.svc.cluster.local"
```

Comment:
This is very much a nitpick, but the variable suggests an IP but that is a FQDN.

If the intention is "something that will resolve to a network resource" my preferred term is usually "host".

(the issue is that in some cases IPs and names are not trivially exchangeable, e.g. something that listens on only some interfaces will work with an IP but might work with a name, and something that uses TLS will work with a name but (preferably) never an IP)

Reply:
Made the change, just for you Dan!